mne/time_frequency/psd.py

```diff
@@ -158,7 +163,12 @@ def compute_epochs_psd(epochs, picks=None, fmin=0, fmax=np.inf, n_fft=256,
                            exclude='bads')
     n_fft, n_overlap = _check_nfft(len(epochs.times), n_fft, n_overlap)
 
-    data = epochs.get_data()[:, picks]
+    if tmin is not None or tmax is not None:
+        time_mask = _time_mask(epochs.times, tmin, tmax)
```

Comment:
but how would you deal with fancy indexing ... imagine you have bad channels, etc.